### PR TITLE
Added new type of edit behaviour

### PR DESCRIPTION
**Pull request by AlexeyGreer**

### CODE REVIEW

File: src/foam/nanos/crunch/edit/OnlyOnboardingEditBehaviour.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * @license
+ * Copyright 2024 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch.edit',
+  name: 'OnlyOnboardingEditBehaviour',
+  extends: 'foam.nanos.crunch.edit.AbstractEditBehaviour',
+
```
Comment (sarthak-marwaha) [inline src/foam/nanos/crunch/edit/OnlyOnboardingEditBehaviour.js:12]: Can you add documentation as to where this needs to be used and why it exists

Comment (AlexeyGreer) [reply src/foam/nanos/crunch/edit/OnlyOnboardingEditBehaviour.js:12]: I guess the name will leave it as a mystery.

File: src/foam/nanos/crunch/edit/OnlyOnboardingEditBehaviour.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  methods: [
+    {
+      name: 'maybeApplyEdit',
+      javaCode: `
+        User user = (User) ((DAO)systemX.get("userDAO")).find(foam.mlang.MLang.EQ(User.ID, ucj.getSourceId()));
+        Group userGroup = (Group) ((DAO) systemX.get("groupDAO")).find(user.getGroup());
+
+        CrunchService crunchService = (CrunchService) systemX.get("crunchService");
+        var GenCapUCJ = crunchService.getJunction(userX, userGroup.getGeneralCapability());
```
Comment (jlhughes) [inline src/foam/nanos/crunch/edit/OnlyOnboardingEditBehaviour.js]: camel case please. 
```suggestion
        var genCapUCJ = crunchService.getJunction(userX, userGroup.getGeneralCapability());
```

**Review by jlhughes — APPROVED** [review]

File: src/foam/nanos/crunch/test/UCJEditBehaviourTest.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
+/**
+ * @license
+ * Copyright 2024 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch.test',
```
Comment (sarthak-marwaha) [inline src/foam/nanos/crunch/test/UCJEditBehaviourTest.js]: I gotta ask, if the capability list only exists in NP does this just fail when it is run in foam?

File: src/foam/nanos/crunch/test/UCJEditBehaviourTest.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        test ( testFullName.getData().toString() != fullNameData.getValue(), "OnlyOnboardingEditBehaviour during onboarding" );
+  
+        // 3. Set phoneNumber(PermissiveEditBehaviour) & address(NullEditBehaviour)
+        testPhoneNumber = crunchService.updateJunctionFor(x, "test.onboarding.phone-number", phoneNumberData, CapabilityJunctionStatus.GRANTED, user, user);
+        testAddress = crunchService.updateJunctionFor(x, "test.onboarding.address", addressData, CapabilityJunctionStatus.GRANTED, user, user);
+
+        // 4. Grant general capability and try to edit all 3, only PermissiveEditBehaviour should be able to
+        var genCapUCJ = crunchService.updateJunctionFor(x, "testOnboarding", null, CapabilityJunctionStatus.GRANTED, user, user);
+
+        testFullName = crunchService.updateJunctionFor(x, "test.onboarding.full-name", fullNameData, CapabilityJunctionStatus.GRANTED, user, user);
```
Comment (sarthak-marwaha) [inline src/foam/nanos/crunch/test/UCJEditBehaviourTest.js]:
```suggestion
        testFullName = crunchService.updateJunctionFor(x, "test.onboarding.full-name", fullNameData2, CapabilityJunctionStatus.GRANTED, user, user);
```

Comment (AlexeyGreer) [reply src/foam/nanos/crunch/test/UCJEditBehaviourTest.js]: At this point its already fullNameData2, then I try to change it back to fullNameData, which I want to fail, and I check it. Not sure what your issue is?

Comment (sarthak-marwaha) [reply src/foam/nanos/crunch/test/UCJEditBehaviourTest.js]: nvm, i misread the code, my bad

File: src/foam/nanos/crunch/test/UCJEditBehaviourTest.js (hunk 3 above)
Comment (jlhughes) [reply src/foam/nanos/crunch/test/UCJEditBehaviourTest.js]: @AlexeyGreer as @sarthak-marwaha mentioned, if the capabilities selected to test: firstname, .... are only onboarding capabilities on nanopay side, then move the test case to nanopay.  net/nanopay/crunch/test/

**Pull request closed without merging**